File: src/main/java/sk/hudak/prco/dto/internal/ProductNewData.java
package sk.hudak.prco.dto.internal;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import sk.hudak.prco.api.EshopUuid;
import sk.hudak.prco.api.Unit;

import java.math.BigDecimal;
import java.util.Optional;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@ToString
public class ProductNewData implements InternalDto {
    //TODO poprehadzovat na optional tie ktore mozu byt null

    private EshopUuid eshopUuid;
    private String url;
    private Optional<String> name;

    private Unit unit;
    private BigDecimal unitValue;
    private Integer unitPackageCount;

    private Optional<String> pictureUrl;

    /**
     * @return true, ak sa podarilo vsetko uspesne vyparsovat(vsetky parametre), inak false
     */
    public boolean isValid() {
        if (url == null || url.trim().length() < 1) {
            return false;
        }
        if (!name.isPresent() || (name.isPresent() && name.get().trim().length() < 1)) {
            return false;
        }
        return eshopUuid != null && unit != null && unitValue != null && unitPackageCount != null;
    }
}

File: src/main/java/sk/hudak/prco/dto/internal/ProductNewData.kt
package sk.hudak.prco.dto.internal

import sk.hudak.prco.api.EshopUuid
import sk.hudak.prco.api.Unit
import java.math.BigDecimal

data class ProductNewData(
        var eshopUuid: EshopUuid? = null,
        var url: String? = null,
        var name: String? = null,
        var unit: Unit? = null,
        var unitValue: BigDecimal? = null,
        var unitPackageCount: Int? = null,
        var pictureUrl: String? = null) : InternalDto {
    //TODO poprehadzovat na optional tie ktore mozu byt null


    /**
     * @return true, ak sa podarilo vsetko uspesne vyparsovat(vsetky parametre), inak false
     */
    val isValid: Boolean
        get() {
            if (url == null || url!!.trim { it <= ' ' }.isEmpty()) {
                return false
            }
            if (name == null || name!!.trim { it <= ' ' }.isEmpty()) {
                return false
            }
            return eshopUuid != null && unit != null && unitValue != null && unitPackageCount != null
        }


}
